GoogleDataTransport/GDTLibrary/GDTStorage.m

```diff
@@ -216,30 +221,28 @@ - (void)appWillForeground:(GDTApplication *)app {
 }
 
 - (void)appWillBackground:(GDTApplication *)app {
-  if (_backgroundID == GDTBackgroundIdentifierInvalid) {
-    _backgroundID = [[GDTApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:^{
-      if (self->_backgroundID != GDTBackgroundIdentifierInvalid) {
-        [[GDTApplication sharedApplication] endBackgroundTask:self->_backgroundID];
-        self->_backgroundID = GDTBackgroundIdentifierInvalid;
-      }
-    }];
-  }
-  dispatch_async(_storageQueue, ^{
-    if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, *)) {
-      NSData *data = [NSKeyedArchiver archivedDataWithRootObject:self
-                                           requiringSecureCoding:YES
-                                                           error:nil];
-      [data writeToFile:[GDTStorage archivePath] atomically:YES];
-    } else {
+  self->_runningInBackground = YES;
+  if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, *)) {
```

Comment:
Was `dispatch_async(_storageQueue, ...)` removed here intentionally? May we bump in to a race condition here (line 116)

Reply:
No, unintentional, good catch.